src/fsharp/NameResolution.fsi

```diff
@@ -541,11 +551,14 @@ val internal ResolveTypeLongIdentInTyconRef         : TcResultsSink -> NameResol
 /// Resolve a long identifier to a type definition
 val internal ResolveTypeLongIdent                   : TcResultsSink -> NameResolver -> ItemOccurence -> FullyQualifiedFlag -> NameResolutionEnv -> AccessorDomain -> Ident list -> TypeNameResolutionStaticArgsInfo -> PermitDirectReferenceToGeneratedType -> ResultOrException<TyconRef>
 
+/// Resolve a long identifier to a type definition with enclosing type instantiations.
+val internal ResolveTypeLongIdentAndEnclosingTypeInst : TcResultsSink -> NameResolver -> ItemOccurence -> FullyQualifiedFlag -> NameResolutionEnv -> AccessorDomain -> Ident list -> TypeNameResolutionStaticArgsInfo -> PermitDirectReferenceToGeneratedType -> ResultOrException<TypeInst * TyconRef>
+
 /// Resolve a long identifier to a field
 val internal ResolveField                           : TcResultsSink -> NameResolver -> NameResolutionEnv -> AccessorDomain -> TType -> Ident list * Ident -> Ident list -> FieldResolution list
 
 /// Resolve a long identifier occurring in an expression position
-val internal ResolveExprLongIdent                   : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TypeNameResolutionInfo -> Ident list -> Item * Ident list
+val internal ResolveExprLongIdent                   : TcResultsSink -> NameResolver -> range -> AccessorDomain -> NameResolutionEnv -> TypeNameResolutionInfo -> Ident list -> Item * Ident list * TypeInst
```

Comment:
This now returns `TypeInst` and wondering if we can make this a little bit clearer? Maybe we should add a new struct type `TyconRefWithTypeInst`:
```
[<Struct>]
type TyconRefWithTypeInst = { TyconRef: TyconRef; TypeInst: TypeInst }
```
And return these instead of having to return `TyconRef` and `TypeInst` in tuples.